### PR TITLE
Port utils to @flow.

### PR DESCRIPTION
**Pull request by marcianx**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -27,21 +28,21 @@ const indexOf = function(list, elem) {
...
-const contains = function(list, elem) {
+const contains = function<T>(list: Array<T>, elem: T) {
     return indexOf(list, elem) !== -1;
 };
 
 /**
  * Provide a default value if a setting is undefined
  */
-const deflt = function(setting, defaultIfUndefined) {
+const deflt = function<T>(setting: T | void, defaultIfUndefined: T): * {
```
Comment (marcianx) [inline src/utils.js]: Unfortunately, I couldn't use `T` as the output type itself as I ran into https://github.com/facebook/flow/issues/5022.

https://flow.org/try/#0MYewdgzgLgBAJgUwGYBsoEYYF4ZIK5jBQCW4APACoB8AFBAlCWAOYBcMFMAPjAG4jE4AGnjIAhnjQBJJAFUwiJMTAI47CgEp2AKhgBvAFAxjMAE4M8psDHqNlzbFhwFFy1TAD8opBOlyFyG5wMOy2TMwA3AYAvlGgkLCKaABM2LgERKRglLRh9urcfALC3r5QMvKuKmocWhz6RibmUJbWeSyOzgFK1Z6lkuX+Ve6hDOFRsQYG8dAwxBAAQmKm6OwARiAgKAhi1jhJGDQugdUiPij0GnHgs-NLpsnrm9u7aQfJR91BZ2IXCFcGIA

File: src/utils.js [hunk 2]
```diff
@@ -55,18 +56,16 @@ const ESCAPE_LOOKUP = {
 
 const ESCAPE_REGEX = /[&><"']/g;
 
-function escaper(match) {
+// Type-safe only if the input string matches ESCAPE_REGEX.
+function escaper(match: string): string {
```
Comment (kevinbarabash) [inline src/utils.js]: It's possible that `escaper` could return `undefined`.

File: src/utils.js [hunk 3]
```diff
@@ -27,21 +28,21 @@ const indexOf = function(list, elem) {
 /**
  * Return whether an element is contained in a list
  */
-const contains = function(list, elem) {
+const contains = function<T>(list: Array<T>, elem: T) {
```
Comment (kevinbarabash) [inline src/utils.js]: The return value for this could be `boolean`.

Comment (marcianx) [reply src/utils.js]: Whoops. I wish flow would catch missing types.

File: src/utils.js (hunk 1 above)
Comment (marcianx) [reply src/utils.js]: Done.

File: src/utils.js (hunk 2 above)
Comment (marcianx) [reply src/utils.js]: Not given the comment I'd added above `escaper()`. It's used only for `escape()`. In fact, I removed this function and just inlined it where it's used.

File: src/utils.js (hunk 3 above)
Comment (kevinbarabash) [reply src/utils.js]: We could start using https://github.com/gajus/eslint-plugin-flowtype#eslint-plugin-flowtype-rules-require-return-type.

Comment (marcianx) [reply src/utils.js]: Yes please!

**Pull request merged**